Mathlib/AlgebraicTopology/SimplicialObject/Coskeletal.lean

```diff
@@ -0,0 +1,116 @@
+extension.
+
+When the ambient category admits right Kan extensions along `(Truncated.inclusion n).op`,
+then when `X` is `n`-coskeletal, the unit of `coskAdj n` defines an isomorphism:
+`isoCoskOfIsCoskeletal : X ≅ (Truncated.cosk n).obj ((truncation n).obj X)`.
+
+TODO: Prove that `X` is `n`-coskeletal whenever a certain canonical cone is a limit cone.
+-/
+
+open Opposite
+
+open CategoryTheory
+
+open CategoryTheory.Limits CategoryTheory.Functor SimplexCategory
+
+universe v u v' u'
+
+namespace CategoryTheory
+
+namespace SimplicialObject
+variable {C : Type u} [Category.{v} C]
+variable (X : SimplicialObject C) (n : ℕ)
+
+namespace Truncated
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion n).op`.-/
+@[simps!]
+def rightExtensionInclusion :
+    RightExtension (Truncated.inclusion n).op
+      ((Truncated.inclusion n).op ⋙ X) := RightExtension.mk _ (𝟙 _)
+
+end Truncated
+
+open Truncated
+
+/-- A simplicial object `X` is `n`-coskeletal when it is the right Kan extension of its restriction
+along `(Truncated.inclusion n).op` via the identity natural transformation. -/
+class IsCoskeletal : Prop where
+  nonempty_isRightKanExtension :
+    Nonempty (IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)))
+
+variable [X.IsCoskeletal n]
+
+/-- If `X` is `n`-cosketal, then `𝟙 ((Truncated.inclusion n).op ⋙ X)` defines a right Kan
+extension of `(Truncated.inclusion.op ⋙ X)` along `(Truncated.inclusion n).op`. -/
+theorem IsCoskeletal.isRightKanExtension :
+    IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)) :=
+  IsCoskeletal.nonempty_isRightKanExtension.some
+
+/-- If `X` is `n`-coskeletal, then `rightExtensionInclusion X n` is a terminal object in the
+category `RightExtension (Truncated.inclusion n).op (Truncated.inclusion.op ⋙ X)`. -/
+noncomputable def IsCoskeletal.isUniversalOfIsRightKanExtension :
+    (rightExtensionInclusion X n).IsUniversal := by
+  have := isRightKanExtension X n
+  apply Functor.isUniversalOfIsRightKanExtension
+
+variable [∀ (F : (SimplexCategory.Truncated n)ᵒᵖ ⥤ C),
+    (SimplexCategory.Truncated.inclusion n).op.HasRightKanExtension F]
+
+/-- There is a map of costructured arrows from `rightExtensionInclusion X n` to the right extension
+of the `n`-truncation of `X` defined by the counit of `coskAdj n`. -/
+noncomputable def Truncated.rightExtensionCosk.hom : Truncated.rightExtensionInclusion X n ⟶
+    RightExtension.mk _
+      ((coskAdj n).counit.app ((Truncated.inclusion n).op ⋙ X)) :=
+  CostructuredArrow.homMk ((coskAdj n).unit.app X) ((coskAdj n).left_triangle_components X)
+
+instance Truncated.isRightKanExtensionCosk : IsRightKanExtension
+    ((Truncated.cosk n).obj ((inclusion n).op ⋙ X))
+    ((coskAdj n).counit.app ((inclusion n).op ⋙ X)) := by
+  unfold Truncated.cosk coskAdj
+  rw [ranAdjunction_counit]
+  exact Functor.instIsRightKanExtensionObjRanAppRanCounit _ _
```

Comment:
As this instance name `Functor.instIsRightKanExtensionObjRanAppRanCounit` was autogenerated by Lean, it would be better to use a syntax like
```lean
  change IsRightKanExtension ...
  infer_instance
```
where in the `change` line, you would put the exact pattern which matches `Functor.instIsRightKanExtensionObjRanAppRanCounit`.

Reply:
Though actually here the "change" wasn't needed!